Src/FluentAssertions/Primitives/StringAssertions.cs

```diff
@@ -660,7 +660,144 @@ public AndConstraint<StringAssertions> ContainEquivalentOf(string expected, stri
+
+            public static bool Twice(int actual) => actual > 2;
+
+            public static bool Thrice(int actual) => actual > 3;
+        }
+
+        public static class LessThan
+        {
+            public static bool Twice(int actual) => actual < 2;
+
+            public static bool Thrice(int actual) => actual < 3;
+        }
+
+        public static class Exactly
+        {
+            public static bool Once(int actual) => actual == 1;
+
+            public static bool Twice(int actual) => actual == 2;
+
+            public static bool Thrice(int actual) => actual == 3;
+        }
+
+        private int CountOccurrences(string expected, StringComparison comparison)
+        {
+            string actual = Subject ?? "";
+            string substring = expected ?? "";
+
+            int count = 0;
+            int index = 0;
+
+            while ((index = actual.IndexOf(substring, index, comparison)) >= 0)
+            {
+                index += substring.Length;
+                count++;
+            }
+
+            return count;
+        }
+
+        /// <summary>
+        /// Asserts that a string contains another (fragment of a) string.
+        /// </summary>
+        /// <param name="expected">
+        /// The (fragment of a) string that the current string should contain.
+        /// </param>
+        /// <param name="because">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <see cref="because" />.
+        /// </param>
+        public AndConstraint<StringAssertions> Contain(string expected, MatchesOccurenceCount predicate, string because = "", params object[] becauseArgs)
+        {
+            Guard.ThrowIfArgumentIsNull(expected, nameof(expected), "Cannot assert string containment against <null>.");
+
+            if (expected.Length == 0)
+            {
+                throw new ArgumentException("Cannot assert string containment against an empty string.", nameof(expected));
+            }
+
+            var actual = CountOccurrences(expected, StringComparison.Ordinal);
+
+            Execute.Assertion
+                .ForCondition(Contains(Subject, expected, StringComparison.Ordinal))
+                .BecauseOf(because, becauseArgs)
+                .FailWith("Expected {context:string} {0} to contain {1}{reason}, but not found.", Subject, expected)
+                .Then
+                .ForCondition(predicate(actual))
+                .BecauseOf(because, becauseArgs)
+                .FailWith(
+                    $"Expected {{context:string}} {{0}} to contain {{1}} {nameof(predicate)} {{reason}}, but found {(actual == 1 ? "1 time" : $"{actual} times")}.",
```

Comment:
Instead of `nameof(predicate)` the message needs to be, for example, `at most 2 times`